packages/flutter_tools/lib/src/devtools_launcher.dart

```diff
@@ -109,8 +108,7 @@ class DevtoolsServerLauncher extends DevtoolsLauncher {
           .transform(utf8.decoder)
           .transform(const LineSplitter())
           .listen(_logger.printError);
-      devToolsUri = await completer.future
-        .timeout(const Duration(seconds: 10));
+      devToolsUrl = await completer.future;
```

Comment:
Can we switch so we are never waiting for the devToolsUrl and instead always determining it in Flutter tool? That would mitigate issues related to the DevTools server not being started yet.
There isn't much of an advantage of DevTools server determining its own port and there is already a code path here where we are passing in our own port to DevTools.

Reply:
do you mean using a set port for DevTools instead of letting the launcher use the first available port starting at 9100?

Reply:
I don't understand this comment thread. Can you elaborate?